src/v2/Apps/Order/Components/SavedAddresses.tsx

```diff
@@ -35,16 +35,18 @@ interface SavedAddressesProps {
   commitMutation?: CommitMutation
   relay: RelayRefetchProp
   addressCount?: number
-  onAddressDelete?: (isLast: boolean) => void
+  onAddressDelete?: (removedAddressId: string) => void
+  onSelectedAddressEdited?: () => void
+  selectedAddress?: string
 }
 // @ts-expect-error STRICT_NULL_CHECK
 type Address = SavedAddresses_me["addressConnection"]["edges"][0]["node"]
 
 const defaultAddressIndex = addressList => {
-  const indexOfDefaultAddress = addressList.findIndex(
-    address => address.node.isDefault
-  )
-  return `${indexOfDefaultAddress > -1 ? indexOfDefaultAddress : 0}`
+  const defaultAddressID = addressList.find(address => address.node.isDefault)
```

Comment:
blocking: What happens if `addressList` is empty? Should we check if `address` in `address.node.isDefault` will have `node`?

Fixing the case where `addressList` is empty will help mitigate errors below as well when we reference `addressList[0]`.

Reply:
Made a small refactoring. Could you please take a look again?